### PR TITLE
[SPARK-45413][CORE] Add warning for prepare drop LevelDB support

### PR DESCRIPTION
**Pull request by Hisoka-X**
### What changes were proposed in this pull request?
This PR add warning deprecated message for use LEVELDB in `spark.history.store.hybridStore.diskBackend` or `spark.shuffle.service.db.backend`.

### Why are the changes needed?
Add warning for LEVELDB users in Spark, so we can remove it safety in the future.

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
exist test.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by mridulm** [comment]
We had already discussed this earlier ... let us default to ROCKSDB for 4.0 and remove support in a later version.

**Review by dongjoon-hyun — DISMISSED** [review]
No, @Hisoka-X  (and cc @LuciferYang )

When I switch to the default (RocksDB) in SPARK-42277, we didn't deprecate LevelDB . 

Switching default is not a deprecation. So, we cannot delete like this without a proper deprecation warning to the existing users.

In addition, `Drop LevelDB` feature should happen in a single release. There is no point to delete partially because Apache Spark binary is still not `LevelDB-free`.

I'd recommend @Hisoka-X to reuse this JIRA ID for deprecation.

**Comment by mridulm** [comment]
For a deployment which is on leveldb currently, this will force use of rocksdb just for this aspect.
IMO, let us remove the support for leveldb from spark together and not partially.

**Comment by mridulm** [comment]
Did not see your message @dongjoon-hyun when I commented - completely agree with your points, and is a super set of what I mentioned :-) )

File: common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java [hunk 1]
```diff
@@ -38,6 +42,7 @@ public static DB initDB(
         switch (dbBackend) {
           case LEVELDB:
             org.iq80.leveldb.DB levelDB = LevelDBProvider.initLevelDB(dbFile, version, mapper);
+            logger.warn("The LEVELDB are deprecated. Please use ROCKSDB instead.");
```
Comment (LuciferYang) [inline common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java:45]: Perhaps it should be `The LEVELDB is deprecated.`?

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for updates.

Could you update the documentation part too? For example, 

https://github.com/apache/spark/blob/94666e9cd63a9471f50ab2fce6c8439c8fd74154/core/src/main/scala/org/apache/spark/internal/config/History.scala#L246

```
- Specifies a disk-based store used in hybrid store; LEVELDB or ROCKSDB.
+ Specifies a disk-based store used in hybrid store; ROCKSDB or LEVELDB (deprecated).
```

https://github.com/apache/spark/blob/94666e9cd63a9471f50ab2fce6c8439c8fd74154/docs/configuration.md?plain=1#L1285

https://github.com/apache/spark/blob/94666e9cd63a9471f50ab2fce6c8439c8fd74154/docs/running-on-yarn.md?plain=1#L920

https://github.com/apache/spark/blob/94666e9cd63a9471f50ab2fce6c8439c8fd74154/docs/spark-standalone.md?plain=1#L338

**Comment by dongjoon-hyun** [comment]
Thank you, @Hisoka-X and all.
Merged to master for Apache Spark 4.0.0.

**Pull request closed without merging**